astra/src/test/java/com/slack/astra/zipkinApi/ZipkinServiceTest.java

```diff
@@ -34,7 +34,7 @@ public class ZipkinServiceTest {
   @BeforeEach
   public void setup() throws IOException {
     MockitoAnnotations.openMocks(this);
-    zipkinService = spy(new ZipkinService(searcher));
+    zipkinService = spy(new ZipkinService(searcher, 20000));
```

Comment:
I think it'd be good to add a test of the max behavior before upstreaming. Now that it's configureable, it'd probably be easier to construct a test case with a service object that has a really small max.